sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -882,6 +894,25 @@ func withProxyEnv() oci.SpecOpts {
 	return oci.Compose(withHttpsProxy, withNoProxy)
 }
 
+// fetchImage returns a `containerd.Image` given an image source.
+func fetchImage(ctx context.Context, source string, client *containerd.Client, registryConfigPath string, fetchCachedImageIfExist bool) (containerd.Image, error) {
+	// Check the containerd image store to see if image exists
+	img, err := client.GetImage(ctx, source)
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			log.G(ctx).WithField("ref", source).Info("Image does not exist, proceeding to pull image from source.")
+		} else {
+			log.G(ctx).WithField("ref", source).Error(err)
+			return nil, err
```

Comment:
I wonder if there are cases where this is not a fatal error? Unlikely I guess.

Reply:
[Doesn't look like it](https://github.com/containerd/containerd/blob/722df11e920cf23563eccef7e2edbcc1f729de29/errdefs/errors.go#L51-L92).

Seems the only time we care to not continue and attempt to pull the image is if it can't be found.